lib/Backend/Lower.h

```diff
@@ -663,6 +664,7 @@ class Lowerer
     IR::Instr*      LowerTry(IR::Instr* instr, bool tryCatch);
     void            EnsureBailoutReturnValueSym();
     void            EnsureHasBailedOutSym();
+    Region *        GetNonNonExceptingFinallyGrandParent(Region *region);
```

Comment:
And rename to GetFirstNonExceptingAncestor, and refactor code in Lowerer::InsertReturnThunkForRegion as discussed offline